src/contracts/MsVault.h

```diff
@@ -0,0 +1,708 @@
+                loggerMsg._type = 5;
+                LOG_INFO(loggerMsg);
+            }
+        }
+        else
+        {
+            state.vaults.set(input.vaultID, locals.vault);
+            loggerMsg._type = 6;
+            LOG_INFO(loggerMsg);
+        }
+    _
+
+    PUBLIC_PROCEDURE_WITH_LOCALS(resetRelease)
+        if (qpi.invocationReward() > RELEASE_RESET_FEE)
+        {
+            qpi.transfer(qpi.invocator(), qpi.invocationReward() - RELEASE_RESET_FEE);
+        }
+        state.totalRevenue += RELEASE_RESET_FEE;
+
+        MSVaultLogger loggerMsg;
+        loggerMsg._contractIndex = CONTRACT_INDEX;
+        loggerMsg._type = 0;
+        loggerMsg.vaultID = input.vaultID;
+        loggerMsg.ownerID = qpi.invocator();
+        loggerMsg.amount = 0;
+        loggerMsg.destination = NULL_ID;
+
+        if (input.vaultID >= MAX_VAULTS)
+        {
+            loggerMsg._type = 1; 
+            LOG_INFO(loggerMsg);
+            return;
+        }
+
+        locals.vault = state.vaults.get(input.vaultID);
+
+        if (!locals.vault.isActive)
+        {
+            loggerMsg._type = 1; 
+            LOG_INFO(loggerMsg);
+            return;
+        }
+
+        locals.io_in.vault = locals.vault;
+        locals.io_in.ownerID = qpi.invocator();
+        isOwnerOfVault(qpi, state, locals.io_in, locals.io_out, locals.io_locals);
+        if (!locals.io_out.result)
+        {
+            loggerMsg._type = 2; 
+            LOG_INFO(loggerMsg);
+            return;
+        }
+
+        locals.fi_in.vault = locals.vault;
+        locals.fi_in.ownerID = qpi.invocator();
+        findOwnerIndexInVault(qpi, state, locals.fi_in, locals.fi_out, locals.fi_locals);
+        locals.ownerIndex = locals.fi_out.index;
+
+        locals.vault.releaseAmounts.set(locals.ownerIndex, 0);
+        locals.vault.releaseDestinations.set(locals.ownerIndex, NULL_ID);
+
+        state.vaults.set(input.vaultID, locals.vault);
+
+        loggerMsg._type = 7;
+        LOG_INFO(loggerMsg);
+    _
+
+    PUBLIC_FUNCTION_WITH_LOCALS(getVaults)
+        output.numberOfVaults = 0;
+        locals.count = 0;
+        for (locals.i = 0; locals.i < 1024; locals.i++)
+        {
+            Vault v = state.vaults.get(locals.i);
+            if (v.isActive)
+            {
+                for (locals.j = 0; locals.j < v.numberOfOwners; locals.j++)
+                {
+                    if (v.owners.get(locals.j) == input.publicKey)
+                    {
+                        output.vaultIDs.set(locals.count, (uint64)locals.i);
+                        output.vaultNames.set(locals.count, v.vaultName);
+                        locals.count++;
+                        break;
+                    }
+                }
+            }
+        }
+        output.numberOfVaults = locals.count;
+    _
+
+    PUBLIC_FUNCTION_WITH_LOCALS(getReleaseStatus)
+        if (input.vaultID >= MAX_VAULTS)
+        {
+            for (locals.i = 0; locals.i < MAX_OWNERS; locals.i++)
```

Comment:
instead of setting all just to 0 you could also add a status to the output that let the user identify that the requested vault is invalid otherwise it could be a user error because he requested wrong vault index

Reply:
the same can apply to the other conditions where the user will not receive a valid result. e.g. inactive vault

Reply:
same could also apply to getBalanceOf, getVaultName